src/legendre.rs

```diff
@@ -0,0 +1,35 @@
+use ff::{Field, PrimeField};
+
+pub trait Legendre: Field {
+    type BasePrimeField: PrimeField;
+
+    // This is (p-1)/2 where p is the modulus of the base prime field
+    fn legendre_exp() -> &'static Vec<u64>;
```

Comment:
I think we could have this as some `const 'static [u64]` and we can print it out and pass it when calling `prime_field_legendre`, that might be an more direct approach? And we can avoid the naming conflict issue.